Maple2.Server.Game/Manager/Field/FieldManager.cs

```diff
@@ -235,6 +239,32 @@ public sealed partial class FieldManager : IDisposable {
         return false;
     }
 
+    public bool TryGetActor(int objectId, [NotNullWhen(true)] out IActor? actor) {
+        if (Players.TryGetValue(objectId, out FieldPlayer? player)) {
+            actor = player;
+            return true;
+        }
+
+        if (Npcs.TryGetValue(objectId, out FieldNpc? npc)) {
+            actor = npc;
+            return true;
+        }
+
+        if (Mobs.TryGetValue(objectId, out FieldNpc? mob)) {
+            actor = mob;
+            return true;
+        }
+
+        if (Pets.TryGetValue(objectId, out FieldPet? pet)) {
+            actor = pet;
+            return true;
+        }
+
+        actor = null;
+
+        return false;
+    }
```

Comment:
This function can be better server as a switch

Reply:
How would you propose using it as a switch? I don't know of a way to identify the target type before locating which dictionary it is in just going by id